src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutes.scala

```diff
@@ -52,73 +53,78 @@ abstract class LeoRoutes(val leonardoService: LeonardoService, val proxyService:
           }
         } ~
         pathPrefix("cluster") {
-          pathPrefix("v2" / Segment / Segment) { (googleProject, clusterName) =>
-            pathEndOrSingleSlash {
-              put {
-                entity(as[ClusterRequest]) { cluster =>
-                  complete {
-                    leonardoService
-                      .processClusterCreationRequest(userInfo, GoogleProject(googleProject), ClusterName(clusterName), cluster)
-                      .map { cluster =>
-                        StatusCodes.Accepted -> cluster
+          pathPrefix("v2" / Segment / Segment) { (googleProject, clusterNameString) =>
+            validateClusterNameDirective(clusterNameString){
```

Comment:
`validateClusterNameDirective` is the main addition, the rest is mostly indentation